### ISSUE TITLE
Possible missing diagnostics

### ISSUE THREAD

**Issue by bschnurr** [microsoft/pyright#10108]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
Possible missing diagnostics based on diagnostic locale string keys.  I might have something configured wrong but i think i have strict mode enabled along with all report errors turned on.

```
{
  "typeCheckingMode": "strict",
  "reportMissingImports": true,
  "reportMissingTypeStubs": true,
  "reportUnknownVariableType": true,
  "reportUnknownArgumentType": true,
  "reportUnknownParameterType": true,
  "reportUnknownMemberType": true,
  "reportMissingTypeArgument": true,
  "reportInvalidTypeVarUse": true,
  "reportCallInDefaultInitializer": true,
  "reportUnnecessaryIsInstance": true,
  "reportUnnecessaryCast": true,
  "reportUnnecessaryComparison": true,
  "reportUnnecessaryContains": true,
  "reportImplicitStringConcatenation": true,
  "reportUndefinedVariable": true,
  "reportUnboundVariable": true,
  "reportInvalidStubStatement": true,
  "reportUnusedCallResult": true,
  "reportUnusedVariable": true,
  "reportDuplicateImport": true,
  "reportWildcardImportFromLibrary": true,
  "reportOptionalSubscript": true,
  "reportOptionalMemberAccess": true,
  "reportOptionalCall": true,
  "reportOptionalIterable": true,
  "reportOptionalContextManager": true,
  "reportOptionalOperand": true,
  "reportTypedDictNotRequiredAccess": true,
  "reportPrivateUsage": true,
  "reportConstantRedefinition": true,
  "reportIncompatibleMethodOverride": true,
  "reportIncompatibleVariableOverride": true,
  "reportOverlappingOverload": true,
  "reportInconsistentConstructor": true,
  "reportPropertyTypeMismatch": true,
  "reportFunctionMemberAccess": true,
  "reportSelfClsParameterName": true,
  "reportImplicitOverride": true,
  "strictListInference": true,
  "strictDictionaryInference": true,
  "strictSetInference": true,
  "enableTypeIgnoreComments": true,
  "reportImportCycles": true,
  "reportMissingSuperCall": true,
  "reportUninitializedInstanceVariable": true,
  "reportUnnecessaryTypeIgnoreComment": true,

}
```

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
# annotatedParamCountMismatch
from typing import Annotated
T = Annotated[int, "meta1", "meta2"]  # two metadata but expected one

# annotatedTypeArgMissing
from typing import Annotated
S = Annotated  # missing type argument and annotation

# assertAlwaysTrue
assert True  # this assert always evaluates to true


# assignmentExprContext
class E:
    x = 5
    y = (x := 10)  # assignment expression not allowed in class body

# assignmentExprInSubscript
my_list = [0]
val = my_list[i := 0]  # assignment expression in subscript (Python 3.10+ only)

# assignmentInProtocol
from typing import Protocol
class Proto(Protocol):
    x = 0  # variables in Protocol must be declared, not assigned


# classVarFirstArgMissing
from typing import ClassVar
class Demo:
    x: ClassVar  # missing type argument

# comparisonAlwaysTrue
if 5 == 5:
    pass  # always evaluates to True

# containmentAlwaysTrue
if "a" in "abc":
    pass  # always True since 'a' is definitely in 'abc'


# coroutineInConditionalExpression
async def coro1(): return 1
async def coro2(): return 2
result = coro1() if True else coro2()  # conditional yields coroutine (always truthy)

# emptyTypeAlias
MyAlias = None  # empty type alias definition


# functionIsNeverCalled
def unused_func():
    print("Never called")


# functionReturnsNever
def error_out():
    raise RuntimeError("Error")  # function never returns normally


# inconsistentReturnStatements
def maybe_value(flag: bool):
    if flag:
        return 1
    # else: missing return (inconsistent return paths)



# invalidDunderName
class D:
    def __notvalid__(self): pass  # invalid dunder name usage

# protocolBaseClass
from typing import Protocol
class MyProto(Protocol): pass
class Impl(MyProto, int):
    pass  # Protocol class cannot derive from non-Protocol base

# protocolUnsafeOverlap
from typing import Protocol
class P1(Protocol): pass
class P2(Protocol): pass
class COverlap(P1, P2): pass  # overlapping Protocols (unsafe overlap)


# typeAliasIllegalExpressionForm
if True:
    Alias = int  # defining type alias in a block (invalid form)

# typeAliasIsRecursiveDirect
from typing import List
TAlias = List['TAlias']  # recursive type alias referencing itself

from typing import TypeVar, Generic
# typeParameterNotUsed
U = TypeVar('U')
class Unused(Generic[U]):
    def __init__(self): 
        pass  # U is defined but not actually used
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

**Comment by erictraut** [comment]
@bschnurr, can you be more specific? I can't tell what bug (if any) you're reporting. Your code block appears to reference internal pyright string IDs. Did you mean to specify documented diagnostic rules?

If you have questions about whether something is a bug, feel free to post to the discussion forum.

If you think that you've identified one or more bugs, please file individual bug reports.

**Comment by bschnurr** [comment]
I was trying to generate a specific test for each diagnostic string. I couldn't get these strings to trigger given the code sample.  I can write them up individually.
